### ISSUE TITLE
LinkCrawlWorker JSON::LD::JsonLdError::CollidingKeywords

### ISSUE THREAD

**Issue by vmstan** [mastodon/mastodon#31314]
### Steps to reproduce the problem

Observed in Sidekiq morgue after merging https://github.com/mastodon/mastodon/pull/31288

<img width="648" alt="image" src="https://github.com/user-attachments/assets/7dcc07a3-dc4c-4ac5-8887-f0a841cae01c">

Errorr Class	JSON::LD::JsonLdError::CollidingKeywords
Error Message	colliding keywords: @id already exists in result

### Expected behaviour

No error

### Actual behaviour

Error

### Detailed description

Status that generated error https://mastodon.social/users/hessenschau/statuses/112915069178628443

### Mastodon instance

vmst.io

### Mastodon version

main

### Technical details

Mastodon	4.3.0-alpha.5+pr-31268
Ruby	3.3.4p94
PostgreSQL	16.3
Redis	7.2.4
OpenSearch	2.14.0
libvips	8.15.2
FFmpeg	7.0.2

**Comment by c960657** [comment]
According to the validator on https://json-ld.org/playground/, the JSON-LD on https://www.hessenschau.de/sport/mehr-sport/spass-statt-wehmut-timo-boll-ist-bei-olympia-megahappy-v1,boll-olympia-124.html is indeed invalid. Depending on how widespread such invalid JSON-LD is, we could gracefully fall back to the old mechanism if the JSON-LD raises an exception. What do you think?

**Comment by c960657** [comment]
@vmstan Do you see a lot of these in your logs?

**Comment by vmstan** [comment]
Yes, and other similar errors. I'd say it's a good 1/3 to 1/2 of my dead queue at the moment.

<img width="1173" alt="Screenshot 2024-08-08 at 1 40 00 PM" src="https://github.com/user-attachments/assets/f0c7642f-14c3-40ae-9e33-2603d3773c69">
